src/tabs/Tab.tsx

```diff
@@ -0,0 +1,202 @@
+/**
+ * Copyright (c) 2018 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import * as React from "react";
+import {Button} from "../forms/button";
+import {ClassNames} from "./ClassNames";
+import {Dropdown, DropdownMenu, DropdownItem, MenuItemType} from "../forms/dropdown";
+
+/**
+ * state for tabs
+ * @param {tabs} List of all Tabs
+ * @param {tabComponent} currently selected tab's component
+ * @param {overflowTab} overflow tab button
+ */
+type TabState = {
+    tabs: TabPane[];
+    tabComponent: React.ReactElement;
+    overflowTab?: TabPane;
+};
+
+/**
+ * props for tabs
+ * @param {index} index of tab in array
+ * @param {tabName} name or title of tab
+ * @param {tabComponent} React element loaded on tab selection
+ * @param {isTabSelected} true if tab is selected
+ * @param {disableTab} true if tab is disabled
+ */
+type TabPane = {
+    index?: number;
+    tabName: string;
+    tabComponent: React.ReactElement;
+    isTabSelected?: boolean;
+    disableTab?: boolean;
+};
+
+/**
+ * Tab Orientation Types
+ * @param {VERTICAL} vertical tabs
+ * @param {HORIZONTAL} Horizontal tabs
+ */
+export enum TabOrientation {
+    VERTICAL = "vertical",
+    HORIZONTAL = "horizontal",
+}
+
+/**
+ * Tab Types
+ * @param {STATIC} User cannot switch between tabs in this static
```

Comment:
Description is bit misleading, please check, if we are can not switch between tabs, then whats the use of other tabs :)

Reply:
Changed the description, it now states -> 'tabs are allowed to be disable enabled conditionally in this type.'